core/blockchain.go

```diff
@@ -2741,13 +2744,27 @@ func (bc *BlockChain) UpdateStakingMetaData(
 // newValidators - the addresses of the newly created validators
 // newDelegations - the map of delegator address and their updated delegation indexes
 func (bc *BlockChain) prepareStakingMetaData(
-	block *types.Block, state *state.DB,
-) (newValidators []common.Address,
-	newDelegations map[common.Address]staking.DelegationIndexes,
-	err error,
+	block *types.Block, stakeMsgs []staking.StakeMsg, state *state.DB,
+) ([]common.Address,
+	map[common.Address]staking.DelegationIndexes,
+	error,
 ) {
-	newDelegations = map[common.Address]staking.DelegationIndexes{}
+	var newValidators []common.Address
+	newDelegations := map[common.Address]staking.DelegationIndexes{}
 	blockNum := block.Number()
+	for _, stakeMsg := range stakeMsgs {
```

Comment:
StakeMsgs are from Normal Transactions which is processed before Staking transaction. So this change is still consistent with the ordering of the processing. Right? No non-deterministic result here.

Reply:
Yes, but you are right that it needs to be thoroughly tested. I would appreciate it if this change spends a bit of time on testnet so I (and others) can verify the behaviour.

Reply:
Looking forward to testing on testnet. Nice work so far.